pkg/storage/ingest/reader.go

```diff
@@ -131,7 +131,9 @@ func (r *PartitionReader) start(ctx context.Context) (returnErr error) {
 	if err != nil {
 		return errors.Wrap(err, "creating service manager")
 	}
-	err = services.StartManagerAndAwaitHealthy(ctx, r.dependencies)
+	// Use context.Background() because we want to stop all dependencies when the PartitionReader stops
+	// instead of stopping them when ctx is cancelled and while the PartitionReader is still running.
+	err = services.StartManagerAndAwaitHealthy(context.Background(), r.dependencies)
```

Comment:
the tradeoff here is that if we get interrupted during startup, we wouldn't respect that and we'd wait for the full startup to finish

not sure if I should try to solve that (maybe with something like `context.AfterFunc()`)

Reply:
If the `PartitionReader` gets terminated (context canceled) while running `PartitionReader.start()`, do we evern call `stopDependencies()` at all? If I remember correctly, `PartitionReader.stop()` will get called only if `PartitionReader.start()` has successfully terminated, that is if it has returned `nil`.

Reply:
we take care of it a few lines above

https://github.com/grafana/mimir/blob/01d32acfb8722f6a9ab37a294b5e35dfaf802206/pkg/storage/ingest/reader.go#L105-L110

Reply:
Oh right, I missed it. LGTM Then.